node/src/peer_tracker.rs

```diff
@@ -81,49 +107,81 @@ impl PeerTracker {
         }
     }
 
-    pub fn connected(&self, peer: PeerId, address: impl Into<Option<Multiaddr>>) {
-        let mut state = self.get(peer);
+    /// Adds a peer as trusted or not.
+    pub fn trusted(&self, peer: PeerId) {
+        self.get(peer).value_mut().trusted = true;
+    }
+
+    /// Inform PeerTracker that there is a new connection for a peer
+    pub fn connected(
+        &self,
+        peer: PeerId,
+        connection_id: ConnectionId,
+        address: impl Into<Option<Multiaddr>>,
+    ) {
+        let mut peer_info = self.get(peer);
 
         if let Some(address) = address.into() {
-            if !state.addrs.contains(&address) {
-                state.addrs.push(address);
+            if !peer_info.addrs.contains(&address) {
+                peer_info.addrs.push(address);
             }
         }
 
-        state.state = PeerState::Connected;
+        peer_info.state = PeerState::Connected;
+        peer_info.connections.push(connection_id);
+
+        // If this is the first connection from the peer
+        if peer_info.connections.len() == 1 {
+            increase_connected_peers(&self.info_tx, peer_info.trusted);
+        }
     }
 
-    pub fn disconnected(&self, peer: PeerId) {
-        let mut state = self.get(peer);
+    /// Inform PeerTracker that a connection of a peer got disconnected
+    ///
+    /// Returns `true` if there are no other connections by the peer
+    pub fn maybe_disconnected(&self, peer: PeerId, connection_id: ConnectionId) -> bool {
```

Comment:
I think I'd prefer having the name convention in peer tracker like:
- `connect`
- `maybe_connect`
- `discover`
- `maybe_discover`
- `disconnect`
- `maybe_disconnect`

those `ed` suffixes for me looks like I'm querying for the information, not changing the state

Reply:
also `identify`

 + we could then drop `is_` in some places

Reply:
or we could keep the convention from the swarm and then have `on_` prefix

Reply:
If we drop the `ed` it will look like we are commanding to connect or disconnect etc. What about `set_` prefix? Like `set_connected`.